Doc/whatsnew/3.10.rst

```diff
@@ -300,6 +300,33 @@ and :data:`types.NotImplementedType` classes, providing a new set
 of types readily interpretable by type checkers.
 (Contributed by Bas van Beek in :issue:`41810`.)
 
+typing
+------
+
+The internally used type cache now supports differentiating types.  For example,
+``Literal[0] == Literal[False]`` previously evaluated to ``True``.  It is now
+``False``.
+(Contributed by Yurii Karabas in :issue:`42345`.)
```

Comment:
Maybe combine the two sections, since they were done together, in the same issue?